### PR TITLE
feat: `api` command should extract source code of api elements

### LINKED ISSUE #1106: `api` command should extract source code of API elements
### Is your feature request related to a problem?

We need to know the source code of API elements when matching them between different versions of an API.

### Desired solution

The `api` command should store the source code (as it exists in the library, incl. comments and custom formatting) in the generated JSON data.

### Possible alternatives (optional)

_No response_

### Screenshots (optional)

_No response_

### Additional Context (optional)

Related to #917.

### PR DESCRIPTION
**Pull request by Aclrian**
Closes #1106.

### Summary of Changes
added an attribute `code` to functions and classes that contain the unformatted source code of the function or class

### Testing Instructions

run `package-parser api -p sklearn -o ./out` and compare the output with the original file that the element contains.
`package-parser annotations -a ../data/api/sklearn__api.json -u ../data/usages/sklearn_usages_counts.json -o ./out/annotations.json` can verify that no error occur if the file is loaded.

### CODE REVIEW

**Comment by lars-reimann** [comment]
> * Does it need tests?

A function like `trim_code` should have tests. However, in this case you might be able to simplify the code by using the [`as_string`](https://pylint.pycqa.org/projects/astroid/en/latest/api/base_nodes.html#astroid.nodes.NodeNG.as_string) method of astroid.

> * Should the data folder be updated? (regarding the [BSD 3-Clause "New" or "Revised" License](https://github.com/scikit-learn/scikit-learn/blob/main/COPYING) from scikit-learn)

For the sake of easy (manual) testing, it makes sense to update the data folder. Just include the license file from scikit-learn as well in the `data/api` folder.

**Comment by Aclrian** [comment]
> However, in this case you might be able to simplify the code by using the [as_string](https://pylint.pycqa.org/projects/astroid/en/latest/api/base_nodes.html#astroid.nodes.NodeNG.as_string) method of astroid.

I compared output of the alternatives and found out that the as_string method ignores comments in the code, but it includes decorators. With a change from fromlineno to lineno decorators would be visible at the current process. Also, it seems that as_string() rearrange lines by removing comments. 

Original `sklearn.datasets._base.load_boston`'s last lines:

```
    return Bunch(
        data=data,
        target=target,
        # last column is target value
        feature_names=feature_names[:-1],
        DESCR=descr_text,
        filename=data_file_name,
        data_module=DATA_MODULE,
    )
```

and what as_string() converts it to:

```
    return Bunch(data=data, target=target, feature_names=feature_names[:-1], DESCR=descr_text, filename=data_file_name, data_module=DATA_MODULE)
```

If the comment line were removed in a new version, there would be a few new lines when we change to as_string().
So, I think we should keep the current procedure.

**Comment by lars-reimann** [comment]
Let's discuss this tomorrow.

File: data/api/LICENSE [hunk 1]
```diff
@@ -0,0 +1,29 @@
+BSD 3-Clause License
```
Comment (lars-reimann) [inline data/api/LICENSE:1]: File should be renamed to something like `scikit-learn license`.

File: package-parser/package_parser/processing/api/_ast_visitor.py [hunk 2]
```diff
@@ -178,6 +202,19 @@ def enter_functiondef(self, function_node: astroid.FunctionDef) -> None:
 
         is_public = self.is_public(function_node.name, qname)
 
+        code = ""
```
Comment (lars-reimann) [inline package-parser/package_parser/processing/api/_ast_visitor.py:205]: Maybe extract this duplicate code to another function.

File: package-parser/tests/processing/api/test_get_code_for_api_element.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import pytest
+
+from package_parser.processing.api._ast_visitor import trim_code
+
+
+@pytest.mark.parametrize(
+    "code_to_pare,expected_code",
+    [
+        (
+            """
```
Comment (lars-reimann) [inline package-parser/tests/processing/api/test_get_code_for_api_element.py:11]: Might be worth adding a function like [`trim_margin`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/trim-margin.html) to make it clear which whitespace is relevant for the test, and which isn't.

**Review by lars-reimann — CHANGES_REQUESTED** [review]

**Comment by lars-reimann** [comment]
Please resolve conversations yourself once you are done with them.

**Review by lars-reimann — APPROVED** [review]

**Pull request merged**